python/lsst/sims/catUtils/utils/SNIaLightCurveGenerator.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import print_function
+import numpy as np
+
+from lsst.sims.catUtils.mixins import SNIaCatalog, PhotometryBase
+from lsst.sims.catUtils.utils import _baseLightCurveCatalog
+from lsst.sims.catUtils.utils import LightCurveGenerator
+
+from lsst.sims.catUtils.supernovae import SNObject, SNUniverse
+from lsst.sims.photUtils import PhotometricParameters, calcGamma
+from lsst.sims.photUtils import Sed, calcMagError_m5, BandpassDict
+
+import time
+
+__all__ = ["SNIaLightCurveGenerator"]
+
+
+class _sniaLightCurveCatalog(_baseLightCurveCatalog, SNIaCatalog, PhotometryBase):
+
+    column_outputs = ["uniqueId", "snid", "raJ2000", "decJ2000",
+                      "cosmologicalDistanceModulus", "redshift", "EBV"]
+
+    _suppressDimSN = False
+
+
+class SNIaLightCurveGenerator(LightCurveGenerator):
+
+    def __init__(self, *args, **kwargs):
+        self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+        self._lightCurveCatalogClass = _sniaLightCurveCatalog
+        self._filter_cat = None
+        self._ax_cache = None
+        self._bx_cache = None
+        self._ax_bx_wavelen = None
+        self.phot_params = PhotometricParameters()
+        self.sn_universe = SNUniverse()
+        self.sn_universe.suppressDimSN = False
+        self.z_cutoff = 1.2
+        super(SNIaLightCurveGenerator, self).__init__(*args, **kwargs)
+
+    class _filterCatalogClass(_sniaLightCurveCatalog):
+        column_outputs = ["uniqueId", "t0"]
+
+    def _light_curves_from_query(self, cat_dict, query_result, grp):
+
+        t_dict = {}
+        gamma_dict = {}
+        m5_dict = {}
+        t_min = None
+        t_max = None
+        for bp_name in cat_dict:
+            self.lsstBandpassDict[bp_name].sbTophi()
+            t_dict[bp_name] = np.array([obs.mjd.TAI for obs in grp
```

Comment:
It seems that in these three loops through grp, we are filtering by all cases where the bandpass is bp_name, so maybe we can group these three loops into a single one